src/units/stimuli.sequence.dragdrop.pas

```diff
@@ -0,0 +1,208 @@
+        //FSample.OnRightDragDrop := @JoinAnimation;
+        //FSample.OnRightDragDrop := @Wrong;
+        //FSample.OnRightDragDrop := @Other;
+        FAnimation.Animate(FSample);
+        FAnimation.Show;
+      end;
+    end;
+
+    for i := low(Comparisons) to high(Comparisons) do
+    begin
+      LItem := TDragDropableItem.Create(self);
+      Comparisons[i].Item := LItem as TObject;
+      FSample.AddTarget(Comparisons[i].Item);
+      LItem.Caption := 'B'+(i+1).ToString;
+      LItem.Parent := Background;
+      LItem.SetOriginalBounds(
+        Comparisons[i].Left,
+        Comparisons[i].Top,
+        Comparisons[i].SquareSide,
+        Comparisons[i].SquareSide);
+      LItem.Show;
+
+      if i = 0 then
+      begin
+        LItem.Caption := 'B'+(i+1).ToString;
+        //LComparison := LItem;
+      end;
+    end;
+    FSample.BringToFront;
+  end;
+
+  //with Grid.RandomPositions do
+  //begin
+  //  for i := low(Comparisons) to high(Comparisons) do
+  //  begin
+  //    LItem := Comparisons[i].Item as TDragDropableItem;
+  //    LItem.SetOriginalBounds(
+  //      Comparisons[i].Left,
+  //      Comparisons[i].Top,
+  //      Comparisons[i].SquareSide,
+  //      Comparisons[i].SquareSide);
+  //  end;
+  //
+  //  for i := low(Samples) to high(Samples) do
+  //  begin
+  //    LItem := Samples[i].Item as TDragDropableItem;
+  //    LItem.Invalidate;
+  //    LItem.SetOriginalBounds(
+  //      Samples[i].Left,
+  //      Samples[i].Top,
+  //      Samples[i].SquareSide,
+  //      Samples[i].SquareSide);
+  //    FAnimation.Animate(LItem);
+  //    FAnimation.Show;
+  //  end;
+  //end;
+end;
+
+procedure TDragDropStimuli.ResetGrid;
+var
+  i : integer;
+begin
+  FSample.OriginalBounds;
+  FSample.Color := clWhite;
+  Grid.RandomizePositions;
+  Grid.RandomizeOrientations;
+
+  with Grid.RandomPositions do begin
+    for i := low(Comparisons) to high(Comparisons) do
+    begin
+      LItem := Comparisons[i].Item as TDragDropableItem;
+      LItem.SetOriginalBounds(
+        Comparisons[i].Left,
+        Comparisons[i].Top,
+        Comparisons[i].SquareSide,
+        Comparisons[i].SquareSide);
+    end;
+
+    for i := low(Samples) to high(Samples) do
+    begin
+      LItem := Samples[i].Item as TDragDropableItem;
+      LItem.Invalidate;
+      LItem.SetOriginalBounds(
+        Samples[i].Left,
+        Samples[i].Top,
+        Samples[i].SquareSide,
+        Samples[i].SquareSide);
+      FAnimation.Animate(LItem);
+      FAnimation.Show;
+    end;
+  end;
+end;
+
+procedure TDragDropStimuli.Stop;
+var
+  i : integer;
+begin
+  with Grid.RandomPositions do begin
+    for i := low(Comparisons) to high(Comparisons) do
+    begin
+      LItem := Comparisons[i].Item as TDragDropableItem;
+      LItem.Destroy;
+    end;
+
+    for i := low(Samples) to high(Samples) do
+    begin
+      LItem := Samples[i].Item as TDragDropableItem;
+      LItem.Destroy;
+      FAnimation.Destroy;
+    end;
+  end;
+end;
```

Comment:
Não é necessário destruir os estímulos aqui. Eles serão destruidos implicitamente no início de cada tentativa. Aqui vejo que basta usar o método `Hide` para escondê-los.

Reply:
Outra coisa. Você deve usar o método `Free` e não o método `Destroy` dos objetos.

Reply:
Certo professor.